### ISSUE TITLE
Masknet replace batchnorm with layernorm

### ISSUE THREAD

**Issue by zippeurfou** [DeepRec-AI/DeepRec#810]
The paper from masknet uses layernorm.
however the code implementation uses batchnorm.

**Comment by StevenShi-23** [comment]
Hi Marc,

Thanks for bringing this up! This is indeed a bug, and we are fixing it.

**Comment by StevenShi-23** [comment]
Hi Marc,

Upon checking, this is not a bug. When applying BatchNorm on the default axis (last dim), BatchNorm reduces to LayerNorm, and since the size of gamma/beta depends on the shape of input tensor, the original implementation is still correct.

However, for the clarity of the code, we updated the example (ref PR #816 ). 

Thanks for the comment!

**Comment by zippeurfou** [comment]
I am not sure I am following see this screenshot.
<img width="876" alt="Screenshot 2023-04-18 at 11 41 50 AM" src="https://user-images.githubusercontent.com/4520383/232830339-ea6010d2-34ea-4191-a5f5-72695c0a5fbf.png">
What am I missing?

**Comment by Duyi-Wang** [comment]
Because your code isn't in trianing.

`tf.layers.batch_normalization()` will call to `class BatchNormalizationBase` 
https://github.com/DeepRec-AI/DeepRec/blob/6bd822e4d05c6b2a005e58342c7661c387b417cb/tensorflow/python/keras/layers/normalization.py#L43
`tf.keras.layers.LayerNormalization() will call to `class LayerNormalization`
https://github.com/DeepRec-AI/DeepRec/blob/6bd822e4d05c6b2a005e58342c7661c387b417cb/tensorflow/python/keras/layers/normalization.py#L898

In LayerNormalization,  mean and var are computed by `nn.moments`
https://github.com/DeepRec-AI/DeepRec/blob/6bd822e4d05c6b2a005e58342c7661c387b417cb/tensorflow/python/keras/layers/normalization.py#L1025
then use `nn.batch_normalization` to get the result.
https://github.com/DeepRec-AI/DeepRec/blob/6bd822e4d05c6b2a005e58342c7661c387b417cb/tensorflow/python/keras/layers/normalization.py#L1040-L1046

It is the same with BN with other features.
https://github.com/DeepRec-AI/DeepRec/blob/6bd822e4d05c6b2a005e58342c7661c387b417cb/tensorflow/python/keras/layers/normalization.py#L643-L652
https://github.com/DeepRec-AI/DeepRec/blob/6bd822e4d05c6b2a005e58342c7661c387b417cb/tensorflow/python/keras/layers/normalization.py#L736-L739
https://github.com/DeepRec-AI/DeepRec/blob/6bd822e4d05c6b2a005e58342c7661c387b417cb/tensorflow/python/keras/layers/normalization.py#L820-L825

But the difference is that when you are not in training, the mean and var of BN will be replaced.
https://github.com/DeepRec-AI/DeepRec/blob/6bd822e4d05c6b2a005e58342c7661c387b417cb/tensorflow/python/keras/layers/normalization.py#L744-L750

**Comment by Duyi-Wang** [comment]
you can add input param `moving_mean_initializer='ones'` which is defaulted to 'zeros' and find output is changed.

**Comment by zippeurfou** [comment]
Thanks it makes sense
